src/util/Repeat.cpp

```diff
@@ -19,24 +19,15 @@
 
 #include "util/Repeat.hpp"
 
-#include <boost/asio/io_context.hpp>
-
 namespace util {
 
-Repeat::Repeat(boost::asio::io_context& ioc) : timer_(ioc)
-{
-}
-
-Repeat::~Repeat()
-{
-    *stopped_ = true;
-}
```

Comment:
Currently without a destructor it is not safe to delete `Repeat` when it is running.